joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/typesolvers/noncaching/JdkJarTypeSolver.scala

```diff
@@ -0,0 +1,171 @@
+package io.joern.javasrc2cpg.typesolvers
+
+import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration
+import com.github.javaparser.symbolsolver.model.resolution.{SymbolReference, TypeSolver}
+import com.github.javaparser.resolution.UnsolvedSymbolException
+import io.joern.javasrc2cpg.typesolvers.JdkJarTypeSolver._
+import javassist.ClassPool
+
+import scala.jdk.CollectionConverters._
+import scala.collection.mutable
+import scala.util.Using
+import org.slf4j.LoggerFactory
+import java.util.jar.JarFile
+import java.io.IOException
+import java.util.jar.JarEntry
+import scala.util.Success
+import scala.util.Try
+import scala.util.Failure
+import io.joern.x2cpg.SourceFiles
+import javassist.CtClass
+import com.github.javaparser.symbolsolver.javassistmodel.JavassistFactory
+import javassist.NotFoundException
+
+class JdkJarTypeSolver private (jdkPath: String) extends TypeSolver {
+
+  private val logger = LoggerFactory.getLogger(this.getClass())
+
+  private var parent: Option[TypeSolver] = None
+  private val classPool                  = new NonCachingClassPool()
+  private val knownClassNames            = mutable.Map[String, String]()
```

Comment:
Please give this a proper name and a description. Since the type is `Map[String, String]` this is not clear from just the types and the current name would be more fit for something like a `Set`.